server/sonar-server/src/test/java/org/sonar/server/computation/step/PersistTestsStepTest.java

```diff
@@ -0,0 +1,344 @@
+    List<CoverageDetail> coverageDetails = Arrays.asList(
+      newCoverageDetail(1, MAIN_FILE_REF_1)
+      );
+    writer.writeCoverageDetails(TEST_FILE_REF_1, coverageDetails);
+
+    sut.execute(new ComputationContext(new BatchReportReader(reportDir), newProjectDto(PROJECT_UUID)));
+
+    FileSourceDto dto = dbClient.fileSourceDao().selectTest(TEST_FILE_UUID_1);
+    assertThat(dto.getCreatedAt()).isEqualTo(now);
+    assertThat(dto.getUpdatedAt()).isEqualTo(now);
+    assertThat(dto.getProjectUuid()).isEqualTo(PROJECT_UUID);
+    assertThat(dto.getFileUuid()).isEqualTo(TEST_FILE_UUID_1);
+    assertThat(dto.getTestData()).hasSize(2);
+
+    assertThat(dto.getTestData()).extracting("name", "coveredFileCount").containsOnly(
+      tuple("name#1", 1),
+      tuple("name#2", 0));
+
+    assertThat(log.logs()).isEmpty();
+  }
+
+  @Test
+  public void insert_all_data_of_a_test() throws Exception {
+    BatchReportWriter writer = new BatchReportWriter(reportDir);
+    writer.writeTests(TEST_FILE_REF_1, Arrays.asList(newTest(1)));
+    writer.writeCoverageDetails(TEST_FILE_REF_1, Arrays.asList(newCoverageDetail(1, MAIN_FILE_REF_1)));
+
+    sut.execute(new ComputationContext(new BatchReportReader(reportDir), newProjectDto(PROJECT_UUID)));
+
+    FileSourceDto dto = dbClient.fileSourceDao().selectTest(TEST_FILE_UUID_1);
+    assertThat(dto.getCreatedAt()).isEqualTo(now);
+    assertThat(dto.getUpdatedAt()).isEqualTo(now);
+    assertThat(dto.getProjectUuid()).isEqualTo(PROJECT_UUID);
+    assertThat(dto.getFileUuid()).isEqualTo(TEST_FILE_UUID_1);
+    assertThat(dto.getTestData()).hasSize(1);
+
+    FileSourceDb.Test test1 = dto.getTestData().get(0);
+    assertThat(test1.getUuid()).isNotEmpty();
+    assertThat(test1.getName()).isEqualTo("name#1");
+    assertThat(test1.getMsg()).isEqualTo("message#1");
+    assertThat(test1.getStacktrace()).isEqualTo("stacktrace#1");
+    assertThat(test1.getStatus()).isEqualTo(TestStatus.FAILURE);
+    assertThat(test1.getExecutionTimeMs()).isEqualTo(1_000);
+    assertThat(test1.getCoveredFileCount()).isEqualTo(1);
+    assertThat(test1.getCoveredFile(0).getCoveredLineList()).containsOnly(1, 2, 3);
+    assertThat(test1.getCoveredFile(0).getFileUuid()).isEqualTo(MAIN_FILE_UUID_1);
+  }
+
+  @Test
+  public void insert_tests_without_coverage_details() throws Exception {
+    BatchReportWriter writer = new BatchReportWriter(reportDir);
+    List<BatchReport.Test> batchTests = Arrays.asList(newTest(1), newTest(2));
+    writer.writeTests(TEST_FILE_REF_1, batchTests);
+
+    sut.execute(new ComputationContext(new BatchReportReader(reportDir), newProjectDto(PROJECT_UUID)));
+
+    FileSourceDto dto = dbClient.fileSourceDao().selectTest(TEST_FILE_UUID_1);
+    assertThat(dto.getFileUuid()).isEqualTo(TEST_FILE_UUID_1);
+    List<FileSourceDb.Test> tests = dto.getTestData();
+    assertThat(tests).hasSize(2);
+    assertThat(tests.get(0).getCoveredFileList()).isEmpty();
```

Comment:
what checks are you thinking about ? there are no coverage details in this case. I've just added a test to check that the 2nd test has no covered file